pkg/vsphere/extraconfig/decode.go

```diff
@@ -334,7 +331,7 @@ func decodeMap(src DataSource, dest reflect.Value, prefix string, depth recursio
 	mapkeys, err := src(prefix)
 	if mapkeys == "" || err != nil {
 		log.Debugf("No value found in data source for maps keys \"%s\"", prefix)
-		return this
+		return reflect.ValueOf(nil)
```

Comment:
This is returned from an unexported function, and I tried to make all of the `decodeX` functions follow the same pattern, i.e. return reflect.ValueOf(nil) in case of error. Returning a nil map would disrupt this pattern, since we can't return for every type that we decode in case of error.

I will add some tests for the error cases.